server.js

```diff
@@ -19,7 +46,64 @@ app.use(express.json())
 
 // Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send('Hello world, welcome to Happy Thoughts API - made by Darya // See this API live in action here: https://happy-thoughts-react-project.netlify.app/')
+});
+
+// getting all of the thoughts and sorting them
+app.get('/thoughts', async (req, res) => {
+  const { page = 1, limit = 5 } = req.query;
+
+  try {
+    const thoughts = await Thought.find({})
+      .limit(limit * 1)
+      .skip((page - 1) * limit)
+      .sort({ createdAt: 'desc' })
+      .exec();
+    // const count = await Thought.countDocuments();
+    res.json({ response: thoughts, success: true })
+  } catch (err) {
+    res.status(400).json({ response: err, success: false });
+  }
+});
+
+// post the message
+app.post('/thoughts', async (req, res) => {
+  const { message, username } = req.body;
+
+  try {
+    const newThought = await new Thought({ message, username: username || 'Anonymous' }).save();
+    res.status(200).json({ response: newThought });
+  } catch (error) {
+    res.status(400).json({ response: error, success: false });
+  }
+});
+
+// update for likes 
+app.post('/thoughts/:thoughtId/like', async (req, res) => {
+  const { thoughtId } = req.params;
+
+  try {
+    // first argument passing id, second what property should be updated
+    const thoughtLiked = await Thought.findByIdAndUpdate(
+      thoughtId, 
+      { $inc: { hearts: 1 } }, { new: true }
+    ); // inc = inscrese, new is options of that method 
+    res.status(200).json({ response: thoughtLiked, success: true })
+  } catch (err) {
+    res.status(400).json({ response: err, success: false });
+  }
+});
+
+// delete thought 
+app.delete('/thoughts/:id', async (req, res) => {
+  const { id } = req.params;
+
+  try {
+    const deleteMessage = await Thought.findOneAndDelete({ _id: id });
+    res.status(200).json({ response: deleteMessage })
+  } catch (err) {
+    res.status(400).json({ response: err, success: false });
+  }
```

Comment:
Really good api and frondend :D easy to read and understand the code, whit discriptive comments! GOOD work :D :D :D